[David Chong Joon Wei] iP

davidchongg opened the pull request:
# Eve 
> “Productivity is never an accident. It is always the result of a commitment to excellence, intelligent planning, and focused effort.” – Paul J. Meyer ([source](https://dansilvestre.com/productivity-quotes/))

Eve is a personal assistant that frees your mind of having to remember things you need to do. It's:
- Text-based
- Easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is
1. Download it from [here](https://github.com/davidchongg/ip/releases/tag/A-Jar).
2. Double-click to launch it.
3. Add your tasks.
4. Sit back and let it manage your tasks for you !!

Andddd it is FREE!

Features:
- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

davidchongg committed:
Change Duke to Eve and implement level 0

davidchongg committed:
Eve.java: Implement level-1 (echo commands)

davidchongg committed:
Eve.java: implement level-2 (add to list and print list)

davidchongg committed:
add Task.java; Eve.java: implement mark as done

davidchongg committed:
add ToDo.java, Deadline.java, Event.java; Eve.java: implement todo, event & deadline tasks

davidchongg committed:
Perform automated text ui testing

davidchongg committed:
Handle exceptions

davidchongg committed:
Handle exceptions (fixed adding issue)

davidchongg committed:
Implement functionality of saving task list to a data file

davidchongg committed:
Merge branch 'branch-Level-7'

davidchongg committed:
Implement functionality of chatbot to understand date and time

davidchongg committed:
Merge branch 'branch-Level-8'

JunXiang26 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+        this.by = by;
+    }
+
+    @Override
+    public String toDataString() {
+        return "D | " + super.getStatusIcon() + " | " + this.description + " | " + this.by;
+    }
```
Acceptable naming convention!

JunXiang26 commented on src/main/java/Eve.java:
```diff
@@ -0,0 +1,221 @@
+        writeToFile();
+    }
+
+    public static void load(String fileName) throws FileNotFoundException {
```
Could specify what your function might be loading

JunXiang26 commented on src/main/java/Eve.java:
```diff
@@ -0,0 +1,221 @@
+        }
+    }
+
+    public static void executeCommand(String[] fullCommand) {
```
Nice work using action verbs to name your function!

JunXiang26 reviewed (commented):
Overall, naming conventions are pretty acceptable!

caxewhy commented on src/main/java/DateTime.java:
```diff
@@ -0,0 +1,34 @@
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+
+public class DateTime {
```
Class name should be a noun representing an object? DateTime might be too ambiguous, perhaps could be renamed to DateTimeParser/DateTimeUtil?

caxewhy commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+        return "D | " + super.getStatusIcon() + " | " + this.description + " | " + this.by;
+    }
+
+    public String toString() {
```
Consider verbs for methods/functions?

caxewhy commented on src/main/java/Eve.java:
```diff
@@ -0,0 +1,221 @@
+public class Eve {
+    private static ArrayList<Task> taskList;
+    private static boolean continueRunning = true;
+    private static final String DATA_FILE = "eve.txt";
```
Good work on the naming of constants!

caxewhy commented on src/main/java/Eve.java:
```diff
@@ -0,0 +1,221 @@
+
+public class Eve {
+    private static ArrayList<Task> taskList;
+    private static boolean continueRunning = true;
```
Boolean variable should sound more like a condition? Maybe consider isRunning?

caxewhy commented on src/main/java/Eve.java:
```diff
@@ -0,0 +1,221 @@
+        }
+    }
+
+    public static void add(String command, String description) throws EventTaskException, DeadlineTaskException{
```
Is method name too generic? Consider addTask()?

caxewhy commented on src/main/java/Eve.java:
```diff
@@ -0,0 +1,221 @@
+        }
+    }
+
+    public static void output(String message) {
```
Method name might be too vague? Consider displayMessage()?

caxewhy commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    @Override
+    public String toDataString() {
```
Does method name clearly convey its purpose? Could consider formatForStorage()?

caxewhy reviewed (commented):
Lets good to go! Just some nits to fix :)

davidchongg committed:
Organize classes using OOP

davidchongg committed:
Merge branch 'branch-A-MoreOOP'

Kaidama97 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,20 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task{
```
Seems to have miss out a spacing before the {?

Kaidama97 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,29 @@
+public class Task {
+    protected String description;
+    private boolean isDone;
```
Good naming convention for boolean variable 👍

Kaidama97 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,74 @@
+        case "E":
+            task = new Event(parts[2], DateTime.parseString(parts[3]), DateTime.parseString(parts[4]));
+            break;
+        default:    // "T"
```
Is there supposed to be spacing between the default and the comment?

Kaidama97 commented on src/main/java/DeleteCommand.java:
```diff
@@ -0,0 +1,26 @@
+    public void execute(ArrayList<Task> taskList, Ui ui, Storage storage) throws EveException {
+        try {
+            Task task = taskList.remove(num - 1);
+            ui.output("Noted. I've removed this task:\n\t" + task
+                    + "\nNow you have " + Integer.toString(taskList.size())
+                    + " tasks in the list.");
```
good indentions for multiple lines 👍

Kaidama97 commented on src/main/java/Eve.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    public void run() {
+        ui.greet();
```
should the method name be something like welcomeMessage()?

Kaidama97 reviewed (commented):
Looks good overall! just some styling to fix 💯

davidchongg committed:
Organize classes into packages

davidchongg committed:
Merge branch 'branch-A-Packages'

davidchongg committed:
Update git ignore to include data file for saving tasks

davidchongg committed:
Merge branch 'add-gradle-support'

davidchongg committed:
Modify build.gradle to specify main class to run

davidchongg committed:
Implement JUnit tests for AddCommand and CommandParser classes

davidchongg committed:
Build to obtain jar file

davidchongg committed:
Add javadoc comments to all the classes and methods

Ditzchann commented on src/main/java/eve/task/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+    @Override
+    public String toDataString() {
+        return "D | " + super.getStatusIcon() + " | " + this.description + " | " + this.by;
+    }
```
Good naming convention!

Ditzchann commented on src/main/java/eve/task/Event.java:
```diff
@@ -0,0 +1,26 @@
+        return "[E]" + super.toString() + " (from: " + this.from.format(DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm"))
+                + "\tto: " + this.to.format(DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm")) + ")";
+    }
+
```
Line breaks were used neatly!

Ditzchann commented on src/main/java/eve/ui/Ui.java:
```diff
@@ -0,0 +1,30 @@
+                + message + "\n"
+                + "-------------------------------------------------------------------------------------\n";
+        System.out.println(fullMessage);
+    }
```
Neat use of line breaks to write multi-lined output

Ditzchann commented on src/main/java/eve/util/Storage.java:
```diff
@@ -0,0 +1,83 @@
+        String[] parts = line.split(" \\| ");
+        for (int i = 0; i < parts.length; i++) {
+            parts[i] = parts[i].trim();
+        }
```
Clear naming of methods!

Ditzchann reviewed (commented):
LGTM!

davidchongg committed:
Clean up code according to coding standards

davidchongg committed:
Implement find command

davidchongg committed:
Merge branch 'branch-A-JavaDoc'

davidchongg committed:
Fix merge conflicts

davidchongg committed:
Fix merge conflicts

davidchongg committed:
Fix minor issues after PR review

davidchongg committed:
Fix code based on checkstyle

davidchongg committed:
Merge branch 'branch-A-Checkstyle'

davidchongg committed:
Implement level 10 gui of chatbot

davidchongg committed:
Merge branch 'branch-Level-10'

davidchongg committed:
Add assertions across critical functions.

Assertions are used to define assumptions about the program
state to be verified at runtime.

Adding assertions will improve the code robustness and help
with debugging.

davidchongg committed:
Improve code quality across the code

The codebase contains inconsistent formatting, redundant functions and
unclear variable names, making it harder to read and maintain.

These issues increases risk of bugs and slow down development.

Refactor code to follow coding standards specified in this module.

davidchongg committed:
Merge pull request #2 from davidchongg/branch-A-Assertions

Add assertions across critical functions.

davidchongg committed:
Merge branch 'master' into branch-A-CodeQuality

davidchongg committed:
Merge pull request #3 from davidchongg/branch-A-CodeQuality

Improve code quality across the code

davidchongg committed:
HelpCommand.java: Remove occurences of ui

davidchongg committed:
Implement sort command

davidchongg committed:
Merge branch 'branch-BCD-Extension'

davidchongg committed:
Improve GUI

davidchongg committed:
Add more personality to Eve

davidchongg committed:
Update user guide

davidchongg committed:
Fix typing errors
